src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -147,6 +147,11 @@ align 16
         jmp     JIT_Box
 NESTED_END JIT_BoxFastMP_InlineGetThread, _TEXT
 
+RELATIVE_PTR_LOAD macro Reg, BaseReg, Offset
```

Comment:
Thank you very much for your comments.

There is a prototype implementation in #12369, which perhaps supports all cases, with except of generics.

Specifically, it doesn't support `System.__Canon[]` arrays - the case requires runtime lookup from the generic type descriptor and signature.

As far as I see, the `JIT_GenericHandleClass` invocation is not generated by JIT, because method tables for `System.__Canon[]` is not considered to be canonical by `th.IsCanonicalSubtype` (and so, by `CEEInfo::embedGenericHandle`).

Is it good idea to add the case to `TypeHandle::IsCanonicalSubtype`?